Patch from upstream; caused due to removal of xf86UDelay()

Fixes http://bugs.gentoo.org/291825
---
From 4e2afb88212582df60bca3ec8c447965a5a66fd5 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 26 Oct 2009 19:43:50 +0000
Subject: s/xf86UDelay/usleep/g

cf. https://bugzilla.redhat.com/show_bug.cgi?id=523800

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/neo_driver.c b/src/neo_driver.c
index ddb6812..e4c60f2 100644
--- a/src/neo_driver.c
+++ b/src/neo_driver.c
@@ -114,6 +114,8 @@ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #include <X11/extensions/xf86dgastr.h>
 #endif
 
+#include <unistd.h>
+
 /* Mandatory functions */
 static const OptionInfoRec *	NEOAvailableOptions(int chipid, int busid);
 static void     NEOIdentify(int flags);
@@ -2489,7 +2491,7 @@ neoRestore(ScrnInfoPtr pScrn, vgaRegPtr VgaReg, NeoRegPtr restore,
      * In some rare cases a lockup might occur if we don't delay
      * here. (Reported by Miles Lane)
      */
-    xf86UDelay(200000);
+    usleep(200000);
     /*
      * Disable horizontal and vertical graphics and text expansions so
      * that vgaHWRestore works properly.
@@ -2502,7 +2504,7 @@ neoRestore(ScrnInfoPtr pScrn, vgaRegPtr VgaReg, NeoRegPtr restore,
      * Sleep for 200ms to make sure that the two operations above have
      * had time to take effect.
      */
-    xf86UDelay(200000);
+    usleep(200000);
     /*
      * This function handles restoring the generic VGA registers.  */
     vgaHWRestore(pScrn, VgaReg,
--
cgit v0.8.2
